### PR TITLE
Include subpages in filtered page results

### PR DESCRIPTION
**Pull request by MizukiTemma**
### Short description

This PR is to fix the problem that subpages are not shown when the filter is applied.

### Proposed changes

- Subpages are now shown too
- Pages are shown even when their root page and/or parent page is filterd out.
- 

### Resolved issues

Fixes: #934

### CODE REVIEW

File: src/cms/templates/pages/page_tree.html
Comment (melegiul) [inline src/cms/templates/pages/page_tree.html:98]: These lines have the purpose to add a drop region after the last root page.
As drag&drop is disabled, when the page filter feature is active, we won't need this template tag here for now.

File: src/cms/templates/pages/page_tree_node.html
Comment (melegiul) [inline src/cms/templates/pages/page_tree_node.html:26]: This template tag is not required, because the `get_children` filter will only return the filtered children, too.
But in this case where we have a partial `TreeQuerySet` only, the `get_children` filter is buggy unfortunately.
So we should fix that over here: https://github.com/Integreat/integreat-cms/blob/38c0601eb4cc94f652fc40d1933adc530d74cab8/src/cms/templatetags/tree_filters.py#L35
This line should actually be:
```
return [child.id for child in node.get_children()]
```
Then the filter should work with partial trees, too.

**Review by melegiul — CHANGES_REQUESTED** [review]
Thank you very much, I have tested your bugfix on chrome and firefox and everything works great
I could just find some minor issues, then this should be ready to merge :tada:

File: src/cms/templates/pages/page_tree.html
Comment (melegiul) [inline src/cms/templates/pages/page_tree.html:94]:
```suggestion
                {% if node == last_root_page %}
```
This was just a typo, right?

File: src/cms/templates/pages/page_tree_node.html
Comment (melegiul) [inline src/cms/templates/pages/page_tree_node.html:26]:
```suggestion
              data-page-children="{{ page|get_children }}">
```
As we fixed now the buggy filter, we can apply it now

File: src/cms/templatetags/page_filters.py
Comment (melegiul) [inline src/cms/templatetags/page_filters.py:38]:
```suggestion
```
Now we can remove this unused tag

File: src/cms/templatetags/page_filters.py
Comment (melegiul) [inline src/cms/templatetags/page_filters.py:97]:
```suggestion
```
This one can now removed safely too

File: src/cms/templatetags/page_filters.py
Comment (melegiul) [inline src/cms/templatetags/page_filters.py:57]: Really cool how you handled the depth calculation of partial trees :muscle: 
Per default only pages from level 0 (the "roots") are shown in the tree,
So we just needed this solution to decide, which pages are the "roots" in a partial tree

**Review by melegiul — CHANGES_REQUESTED** [review]
Great, now beside my comments only thing left is doing a rebase and commit squashing would be nice too.
If you need help resolving the conflicts please let me know.

**Review by melegiul — APPROVED** [review]
Awesome :tada:

**Comment by timoludwig** [comment]
Please wait with merging this PR until I squashed all commits, thanks!

**Comment by timoludwig** [comment]
@MizukiTemma Thank for you PR, you can now merge it :tada:

For the future: Please use the command `git rebase develop` instead of `git merge develop` when you want to include the recent changes of the develop branch in your feature branch like described in [our documentation](https://integreat.github.io/integreat-cms/git-flow.html#update-feature-branch). This makes the git history a bit cleaner and makes it easier to squash the commits later on.
Also, please change the subjects of your commit messages so that they describe the solution to the problem, not the problem itself. E.g., If the problem is that the subpages are not shown, the commit that fixes that issue should be named "Include subpages in filtered page results" or similar.

**Comment by timoludwig** [comment]
@MizukiTemma This exactly what I'm talking about, please don't merge the develop branch into your feature branch, but use rebasing instead :slightly_smiling_face: 
I will rebase this branch now again before merging.

**Pull request merged**
